local_pathfinding/objectives.py

```diff
@@ -24,66 +53,313 @@ def motionCost(self, s1: ob.SE2StateSpace, s2: ob.SE2StateSpace) -> ob.Cost:
+        self.wind_direction = math.radians(wind_direction)
 
     # This objective function punishes the boat for going up/downwind
     def motionCost(self, s1: ob.SE2StateSpace, s2: ob.SE2StateSpace) -> ob.Cost:
-        return ob.Cost()
+        """Generates the cost associated with the upwind and downwind directions of the boat in
+           relation to the wind
+
+        Args:
+            s1 (SE2StateInternal): The starting point of the local start state
+            s2 (SE2StateInternal): The ending point of the local goal state
+
+        Returns:
+            float: The cost of going upwind or downwind
+        """
+        distance = ((s2.getY() - s1.getY()) ** 2 + (s2.getX() - s1.getX()) ** 2) ** 0.5
+        boat_direction_radians = math.atan2(s2.getX() - s1.getX(), s2.getY() - s1.getY())
+
+        if WindObjective.is_upwind(self.wind_direction, boat_direction_radians):
+            return UPWIND_MULTIPLIER * distance
+        elif WindObjective.is_downwind(self.wind_direction, boat_direction_radians):
+            return DOWNWIND_MULTIPLIER * distance
+        else:
+            return 0.0
 
     @staticmethod
-    def _is_upwind(wind_direction_radians: float, boat_direction_radians: float) -> bool:
-        raise NotImplementedError
+    def is_upwind(wind_direction: float, boat_direction: float) -> bool:
+        """Determines whether the boat is upwind or not and its associated cost
+
+        Args:
+            wind_direction (float): The true wind direction (radians). [-pi, pi)
+            boat_direction (float): The direction of the boat (radians). [-pi, pi)
+
+        Returns:
+            bool: The cost associated with the upwind direction
+        """
+        theta_min = wind_direction - UPWIND_MAX_ANGLE_RADIANS
+        theta_max = wind_direction + UPWIND_MAX_ANGLE_RADIANS
+
+        return WindObjective.is_angle_between(theta_min, boat_direction, theta_max)
 
     @staticmethod
-    def _is_downwind(wind_direction_radians: float, boat_direction_radians: float) -> bool:
-        raise NotImplementedError
+    def is_downwind(wind_direction: float, boat_direction: float) -> bool:
+        """Generates the cost associated with the downwind direction
+
+        Args:
+            wind_direction (float): The true wind direction (radians). [-pi, pi)
+            boat_direction_radians (float)): The direction of the boat (radians). [-pi, pi)
+
+        Returns:
+            bool: The cost associated with the downwind direction
+        """
+        downwind_wind_direction = (wind_direction + math.pi) % (2 * math.pi)
+
+        theta_min = downwind_wind_direction - DOWNWIND_MAX_ANGLE_RADIANS
+
+        theta_max = downwind_wind_direction + DOWNWIND_MAX_ANGLE_RADIANS
+
+        return WindObjective.is_angle_between(theta_min, boat_direction, theta_max)
+
+    @staticmethod
+    def is_angle_between(first_angle: float, middle_angle: float, second_angle: float) -> bool:
```

Comment:
for Jamen: Does this work for unbounded angles? the `is_upwind` function could give unbounded angles.

Reply:
Made `is_angle_between()` handle un bounded angles, because of functions like `is_upwind()`. In `is_upwind()`, we could have `wind_angle` = pi (or nearly pi), then when we add on the `HIGHEST_UPWIND_ANGLE_RADIANS` we could pass an unbounded angle to `is_angle_between()`.